File: src/week1/kim_sung_hyun/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String,Integer> map = new HashMap<>();

        for(int i=0; i<clothes.length; i++)
        {
            String clothe = clothes[i][1];
            map.put(clothe,map.getOrDefault(clothe,0)+1);
        }


        Iterator<Map.Entry<String,Integer>> entries = map.entrySet().iterator();
        while(entries.hasNext())
        {
            Map.Entry<String,Integer> entry = entries.next();
            answer *=entry.getValue()+1;
        }

        //clothes.size + 각 옷 개수를 곱해야함

        return answer-1;
    }
}

File: src/week1/kim_ji_il/위장.cpp
/*  # 1번 풀이 (통과)
- 문제 : [위장]
- 접근 방식 : 처음에는 모든 조합을 구할 생각에 어렵게 생각 했는데, 개수만 계산해주면 되는 문제임을 알고는 쉽게 풀 수 있었다.
             hash_map에 종류별 옷 개수를 count 해주고, 선택안함 포함해서 옷 종류별 개수 +1 를 곱해주면 정답 ( 마지막에 모두 선택 안한 경우 : -1)
- 시간 복잡도 : O(n)
*/

#include <string>
#include <vector>
#include <unordered_map>

using namespace std;    
    
int solution(vector<vector<string>> clothes) {
    int answer = 0;
    
    unordered_map<string,int> hash_map;
    
    for(size_t i=0;i<clothes.size();i++){
        hash_map[clothes[i][1]]++;
    }
    
    answer=1;
    for(auto elem:hash_map){
        answer *= (elem.second+1);
    }
   
    return answer-1;
}